source/write/transactions.txt

```diff
@@ -0,0 +1,194 @@
+         in the API documentation.
+       |
+       | Raises an exception if the options are misconfigured, if there are network or other transient failures, or if there 
+         are other errors such as a session with a transaction already in progress. If an error is returned with the ``TransientTransactionError`` label, 
+         you can end the transaction and then retry it with the expectation that it will succeed. 
+       |
+       | To learn more about this method, see the :manual:`startTransaction()
+         </reference/method/Session.startTransaction/>` guide in the {+mdb-server+} manual.
+
+   * - ``commit_transaction()``
+     - | Commits the active transaction on the current client session.
+       |
+       | Raises an exception if options are misconfigured, if there are network or other transient failures,
+         or if there are other errors such as a session with no transaction in progress. If an error is returned with the ``UnknownTransactionCommitResult`` label,
+         you can end the transaction and then retry it with the expectation that it will succeed when the committed transaction satisfies the set write concern. 
+       |
+       | To learn more about this method, see the :manual:`commitTransaction()
+         </reference/method/Session.commitTransaction/>` guide in the {+mdb-server+} manual. 
+
+   * - ``abort_transcaction()``
+     - | Ends the active transaction on the current client session. 
+       |
+       | Raises an exception if the options are misconfigured or if there are other errors such as 
+         a session with no transaction in progress.
+       |
+       | To learn more about this method, see the :manual:`abortTransaction()
+         </reference/method/Session.abortTransaction/>` guide in the {+mdb-server+} manual.
+
+.. tip:: 
+   
+   The ``mongocxx::client_session`` class also provides methods to retrieve and modify session properties.
+   To learn more, see `mongocxx::client_session <{+api+}/classmongocxx_1_1v__noabi_1_1client.html>`__ in the API documentation. 
+
+The following code shows how to use the core API to perform a transaction that inserts documents into collections in the ``sample_mflix`` database. 
+When run, the code performs the following actions: 
+
+1. Starts a session from the client using the ``start_session()`` method.
+#. Creates an option object to set the write preference for the transaction operations. 
+#. Calls the ``start_transaction()`` method to start a transaction, passing in the option object as an argument.
+#. Attempts operations to insert documents into collections in the ``sample_mflix`` database. 
+   If an operation encounters an error, the transaction is aborted and
+   then retried if the error has the label ``TransientTransactionError``.  
+#. Commits the active transaction using the ``commit_transaction()`` method. If the commit encounters an error with the label ``UnknownTransactionCommitResult``, the commit is retried. 
```

Comment:
S: Same as above - consider moving after the code?

Reply:
Same response as above